gs.js

```diff
@@ -122,9 +127,65 @@ class GlobalStorage extends lib.StorageProvider {
   ) {
     return this.nextId++;
   }
+
+  // Get system suffix for given id
+  //   id - <common.Uint64>
+  //
+  // Returns: <common.Uint64>
+  getSystemSuffix(id) {
+    return common.Uint64.and(id, this.systemBitmask);
+  }
+
+  // Check whether data with given id is stored on this system
+  //   id - <common.Uint64>
+  //
+  // Returns: <boolean>
+  curSystem(id) {
+    return common.Uint64.cmp(this.getSystemSuffix(id), this.systemSuffix);
```

Comment:
`Uint64.cmp()` returns a number, not a boolean, you should compare it with 0 here to check if two numbers are equal:
```suggestion
    return common.Uint64.cmp(this.getSystemSuffix(id), this.systemSuffix) === 0;
```